db/version_set.cc

```diff
@@ -565,10 +565,15 @@ Status Version::GetTableProperties(std::shared_ptr<const TableProperties>* tp,
     s = ioptions->env->NewRandomAccessFile(
         *fname, &file, vset_->env_options_);
   } else {
-    s = ioptions->env->NewRandomAccessFile(
+    std::string new_fname =
         TableFileName(vset_->db_options_->db_paths, file_meta->fd.GetNumber(),
-                      file_meta->fd.GetPathId()),
-        &file, vset_->env_options_);
+                      file_meta->fd.GetPathId());
+    s = ioptions->env->LoadPrivateMetadata(new_fname, file_meta->priv_meta);
+    if (!s.ok()) {
+      return s;
+    }
+    s = ioptions->env->NewRandomAccessFile(new_fname, &file,
+                                           vset_->env_options_);
```

Comment:
We can, but LoadPrivateMetadata has nothing to do with NewRandomAccessFile. The reason why we load metadata here is because it is the first place when recovering the database that we look at tables one by one - and we happen to read from them to check their footer.

I think that we would be mixing two different functionalities on one single function, which we might have to separate in the future if things change. But I can do that if yo think that one single call to Env is worth it :)